charts/akeyless-api-gateway/templates/_helpers.tpl

```diff
@@ -416,17 +416,33 @@ Check metrics configuration Secret
 {{- end }}
 
 {{- define "health_check_path" -}}
-{{- if not (regexMatch "^[0-9]+(\\.[0-9]+){2}$" (include "version" .)) -}}
-/health
-{{- else -}}
-{{- $parts :=  split "." (include "version" .) -}}
-{{- $major := index $parts "_0" | int }}
-{{- $minor := index $parts "_1" | int }}
-{{- $patch := index $parts "_2" | int }}
-{{- if or (gt $major 4) (and (eq $major 4) (or (gt $minor 10) (and (eq $minor 10) (ge $patch 0)))) -}}
+{{- if or (regexMatch "^[0-9]+(\\.[0-9]+){2}$" (include "version" .)| not) (semverCompare ">=4.10.0" (include "version" .)) -}}
 /health
 {{- else -}}
 /
 {{- end }}
 {{- end }}
-{{- end }}
\ No newline at end of file
+
+{{- define "cache_enable_tls" -}}
+{{ and .Values.cache .Values.cache.tls (or .Values.cache.tls.autoGenerated (ne .Values.cache.tls.existingSecretName "")) }}
```

Comment:
this condition is not accurate we need to look at clusterCache.enabled and .Values.cache.tls is not a value ?
please fix

Reply:
.Values.cache.tls is indeed not a value but I still need to traverse it otherwise I might get a nil pointer reference on .Values.cache.tls.autoGenerated 

clusterCache.enabled is not relevant here since the configuration of the redis cache is encapsulated:
```
            {{- if or (eq (include "akeyless-api-gw.adminAccessUidExist" .) "true") ( .Values.cachingConf.clusterCache.enabled) }}
            - name:  GATEWAY_CLUSTER_CACHE
              value: enable
            - name: ENABLE_CACHE_TLS
              value: {{ include "cache_enable_tls" . | quote }}
```

Reply:
i have no problem with the logic/condition but the ENABLE_CACHE_TLS and it's config should be under cluster cache configuration to prevent user mistakes, with that said i think product should decide where it should be located @renanaAkeyless